x/auth/client/cli/tx_multisign.go

```diff
@@ -195,6 +201,155 @@ func makeMultiSignCmd() func(cmd *cobra.Command, args []string) (err error) {
 	}
 }
 
+func GetMultiSignBatchCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "multisign-batch [file] [name] [[signature-file]...]",
+		Short:  "Assemble multisig transactions",
+		Long:   `Assemble a batch of multisig transactions generated by batch sign command.`,
+		PreRun: preSignCmd,
+		RunE:   makeBatchMultisignCmd(),
+		Args:   cobra.MinimumNArgs(3),
+	}
+
+	cmd.Flags().Bool(flagNoAutoIncrement, false, "disable sequence auto increment")
+	cmd.Flags().String(
+		flagMultisig, "",
+		"Address of the multisig account on behalf of which the transaction shall be signed",
+	)
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func makeBatchMultisignCmd() func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		clientCtx := client.GetClientContextFromCmd(cmd)
+		clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+		if err != nil {
+			return err
+		}
+
+		txCfg := clientCtx.TxConfig
+		txFactory := tx.NewFactoryCLI(clientCtx, cmd.Flags())
+		if txFactory.SignMode() == signingtypes.SignMode_SIGN_MODE_UNSPECIFIED {
+			txFactory = txFactory.WithSignMode(signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON)
+		}
+
+		var infile = os.Stdin
+		if args[0] != "-" {
+			infile, err = os.Open(args[0])
+			if err != nil {
+				return err
+			}
+		}
+		scanner := authclient.NewBatchScanner(txCfg, infile)
+
+		kb := clientCtx.Keyring
+		multisigInfo, err := kb.Key(args[1])
```

Comment:
There are some parts (e.g. here and below) that are shared with makeMultiSignCmd. Would there be a way to factorize that?

Reply:
Refactoring should ensue in a separate PR.